osbuild/util/markup.py

```diff
@@ -0,0 +1,28 @@
+"""Wrapper module for output formatting."""
+
+import sys
+
+from typing import Dict
+
+
+class VT:
+    """Video terminal output, disables formatting when stdout is not a tty."""
+
+    escape_sequences: Dict[str, str] = {
+        "reset": "\033[0m",
+
+        "bold": "\033[1m",
+
+        "red": "\033[31m",
+        "green": "\033[32m",
+    }
+
+
+    def __getattr__(self, name: str) -> str:
+        if not sys.stdout.isatty():
+            return ""
```

Comment:
This runs an ioctl-syscall each time you fetch an escape-sequence, kinda working against the concept of IO buffering.

How about calling `isatty()` in the constructor of `VT`, and then creating either a VT object that supports escape sequences, or one that doesn't? This would also allow overriding the mode in the future, because if you pipe output into `less` or similar tools, `isatty()` returns false, even though you actually want coloring/etc.

Regardless, I think we can do that in a followup.

Reply:
Agree, moved it to a property on the object. We can add overridable behaviour later.